server/app.js

```diff
@@ -29,22 +29,28 @@ const io = socketio(server, {
     origin: "*",
   },
 });
+app.set('socketio', io);
+io.on("connection",(socket) => {
+   
+  socket.on('JoinConversation', async({ userId,conversationId},cb) => {
+
+   const{error}=await addUser({id:socket.id,userId,conversationId})
 
-io.on("connection", (socket) => {
-  socket.on('JoinConversation', ({ userProfileId,conversationId },cb) => {
-    const{error}=addUser({id:socket.id,userProfileId,conversationId})
     if(error)return cb(error)
+
     socket.join(conversationId);
+
     cb()
   }) 
-
-  socket.on('chatMessage', (message) => {
+  socket.on('chatMessage', (message,cb) => {
     const user=getUser(socket.id)
-    io.to(user.conversationId).emit('message', {userProfileId,message});
+    io.to(socket.id).emit('message', {senderId:user.userId,text:message});
+    cb()
   });
 
   socket.on('disconnect',()=>{
-   removeUser(socket.id);
+   const user=removeUser(socket.id);
```

Comment:
Same here, I would really like to see you go over and format your back-end code. Adding whitespace where appropriate.

Reply:
Hi Ethan, Thank you for your review. As we discussed before, the `Prettier` extension did not work at the backend for some reason. but I will pay attention in the future.